pdf2docx/common/Element.py

```diff
@@ -172,43 +178,45 @@ def get_main_bbox(self, e, threshold:float=0.95):
         """
         bbox_1 = self.bbox
         bbox_2 = e.bbox if hasattr(e, 'bbox') else fitz.Rect(e)
-        
+
         # areas
         b = bbox_1 & bbox_2
-        if not b: return None # no intersection
-
-        a1, a2, a = bbox_1.get_area(), bbox_2.get_area(), b.get_area()        
+        if b.is_empty: return None # no intersection
 
         # Note: if bbox_1 and bbox_2 intersects with only an edge, b is not empty but b.get_area()=0
         # so give a small value when they're intersected but the area is zero
+        a1, a2, a = bbox_1.get_area(), bbox_2.get_area(), b.get_area()
         factor = a/min(a1,a2) if a else 1e-6
         return bbox_1 | bbox_2 if factor >= threshold else None
 
 
     def vertically_align_with(self, e, factor:float=0.0, text_direction:bool=True):
-        '''Check whether two Element instances have enough intersection in vertical direction, i.e. perpendicular to reading direction.
-        
+        '''Check whether two Element instances have enough intersection in vertical direction,
+        i.e. perpendicular to reading direction.
+
         Args:
             e (Element): Object to check with
-            factor (float, optional): Threshold of overlap ratio, the larger it is, the higher probability the two bbox-es are aligned.
-            text_direction (bool, optional): Consider text direction or not. True by default, from left to right if False.
+            factor (float, optional): Threshold of overlap ratio, the larger it is, the higher
+                probability the two bbox-es are aligned.
+            text_direction (bool, optional): Consider text direction or not.
+                True by default,from left to right if False.
```

Comment:
I don't really understand what this means "True by default, from left to right if False".

So if the `text_direction` is `True` then we do consider text direction - okay, but if it is `False` then we don't consider text directions, however when I read this it seems like `False` means we consider a left to right text direction as it says "from left to right if False". I'm confused!

Reply:
I didn't read any contradictions between your thoughts -> don't consider text directions -> ignore real text directions -> use default text directions -> the most common case, horizontal, i.e., from left to right.

Appreciated if you help a precise wording.

Reply:
Per my understanding, don't consider text directions, means to use default text direction, which is from left to right.

Reply:
I suppose the bit I don't understand then is if `True` then what is the text direction? Basically:

`False` = from left to right
`True` = ?

Also if we are `False` then we do consider the text direction don't we (left to right)? Which is why `text_direction (bool, optional): Consider text direction or not.` confuses me!

Reply:
I probably understand your confusion. When `False`, the words `from left to right` does not mean to text direction, but the default direction for "horizontal". I should just keep `True by default` and delete the rest.

![image](https://github.com/ArtifexSoftware/pdf2docx/assets/13885442/45d7dd59-9953-40d5-9e43-21f705c1939e)

Reply:
Okay - yes please just delete to avoid the confusion.